### PR TITLE
#144 pl/i support for debug and z unit testcase build

### PR DESCRIPTION
**Pull request by flpet-ibm**

### CODE REVIEW

File: languages/PLI.groovy [hunk 1]
```diff
@@ -23,15 +23,27 @@ buildUtils.createLanguageDatasets(langQualifier)
...
+	langQualifier = "pli_test"
+	buildUtils.createLanguageDatasets(langQualifier)
+}
+
 // iterate through build list
 sortedList.each { buildFile ->
 	println "*** Building file $buildFile"
 
+	// Check if this a testcase
+	isZUnitTestCase = (props.getFileProperty('pli_testcase', buildFile).equals('true')) ? true : false
```
Comment (dennis-behm) [inline languages/PLI.groovy:36]: Hi @flpet-ibm,

I am missing the definition of `pli_testcase` in the `sample/application-conf/file.properties` file in your PR. Please add that like the `cobol_testcase` and also update the README for it.

File: languages/PLI.groovy [hunk 2]
```diff
@@ -98,6 +110,7 @@ def createPLIParms(String buildFile, LogicalFile logicalFile) {
 	def cics = props.getFileProperty('pli_compileCICSParms', buildFile) ?: ""
 	def sql = props.getFileProperty('pli_compileSQLParms', buildFile) ?: ""
 	def errPrefixOptions = props.getFileProperty('pli_compileErrorPrefixParms', buildFile) ?: ""
+	def compileDebugParms = props.getFileProperty('pli_compileDebugParms', buildFile)
```
Comment (dennis-behm) [inline languages/PLI.groovy:113]: This build property does not exist in the sample as of today - 'samples/application-conf/PLI.properties' .

File: languages/PLI.groovy [hunk 3]
```diff
@@ -214,11 +244,19 @@ def createLinkEditCommand(String buildFile, LogicalFile logicalFile, String memb
 
 
 	// define the MVSExec command to link edit the program
+	if(isZUnitTestCase)
+	   parms = "DYNAM(DLL)"
```
Comment (dennis-behm) [inline languages/PLI.groovy:248]: While we allow the parms to be file level overwrites, we should manage this property as such in the `samples/application-conf/PLI.properties` : 

`pli_linkEditParms=DYNAM(DLL) :: **/testcase/*.pli`
(https://github.com/IBM/dbb-zappbuild/blob/development/samples/application-conf/PLI.properties#L37)

**Review by dennis-behm — CHANGES_REQUESTED** [review]
Thanks for the contribution Flemming, I have commented on a couple of sections of your implementation. Please feel free to reach out to me.

File: samples/application-conf/file.properties [hunk 4]
```diff
@@ -17,18 +17,17 @@ dbb.scriptMapping = ZunitConfig.groovy :: **/*.bzucfg
...
 
 #
-# General file level overwrites through DBB Build Properties 
-# isCICS = true :: **/cobol/member.cbl
-# isSQL = true :: **/cobol/member.cbl
-# isMQ = true :: **/cobol/member.cbl
-
-# 
-# Please check for available file property overwrites within samples/application-conf/README.md
+# Skip creating a load module for these programs as they will be statically linked to other programs
```
Comment (dennis-behm) [inline samples/application-conf/file.properties:20]: @flpet-ibm Hi Flems, `samples/application-conf/file.properties` should not contain any specific configuration for MortgageApplication. 

I think you have combined the MortgageSample with the **Template** in sample/application-conf/

**Comment by flpet-ibm** [comment]
I was not aware that I had pushed the changes to the repo 😂 still work in progress 

Sent from my iPhone

> On 26 Nov 2021, at 18.13, Dennis Behm ***@***.***> wrote:
> 
> ﻿
> This Message Is From an External Sender
> This message came from outside your organization.
> @dennis-behm commented on this pull request.
> 
> In samples/application-conf/file.properties:
> 
> > @@ -17,18 +17,17 @@ dbb.scriptMapping = ZunitConfig.groovy :: **/*.bzucfg
>  dbb.scannerMapping = ZUnitConfigScanner :: **/*.bzucfg
>  
>  #
> -# General file level overwrites through DBB Build Properties 
> -# isCICS = true :: **/cobol/member.cbl
> -# isSQL = true :: **/cobol/member.cbl
> -# isMQ = true :: **/cobol/member.cbl
> -
> -# 
> -# Please check for available file property overwrites within samples/application-conf/README.md
> +# Skip creating a load module for these programs as they will be statically linked to other programs
> @flpet-ibm Hi Flems, samples/application-conf/file.properties should not contain any specific configuration for MortgageApplication.
> 
> I think you have combined the MortgageSample with the Template in sample/application-conf/
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
Medmindre andet er angivet ovenfor: / Unless Otherwise Stated Above:
IBM Danmark ApS
Prøvensvej 1
2605 Brøndby, Danmark
CVR nr.: 65305216

File: build-conf/README.md [hunk 5]
```diff
@@ -158,15 +158,21 @@ pli_srcOptions | BPXWDYN creation options for creating 'source' type data sets
 pli_loadDatasets | Comma separated list of 'load module' type data sets
 pli_loadOptions | BPXWDYN creation options for 'load module' type data sets
 pli_tempOptions | BPXWDYN creation options for temporary data sets
+pli_test_case_srcPDS | Dataset to move COBOL test source files to from USS
```
Comment (dennis-behm) [inline build-conf/README.md:161]: COBOL -> PLI

**Review by dennis-behm — CHANGES_REQUESTED** [review]
@flpet-ibm Just found a typo in the README :-) Can you please fix it. I will merge it into the develop branch right after. Thanks

**Review by dennis-behm — APPROVED** [review]

**Pull request merged**